### PR TITLE
fix(ng-add): undefined project in assets and tsconfig paths

### PR DESCRIPTION
**Pull request by eranshmil**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI-related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Issue Number

Somewhat related to the following issue.

Issue Number: #115 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
See below [comment](https://github.com/ng-doc/ng-doc/pull/160#issuecomment-2156037058) for more information.

### CODE REVIEW

**Comment by skoropadas** [comment]
Hey, thank you for the PR! Please give me some time to review it (as I'm currently busy with moving to a new house)

**Comment by skoropadas** [comment]
Checked the PR, and I'm curious why do we need the `project` parameter? Initially, I thought this ticket was about a lack of documentation for monorepos.

So if we try to install NgDoc for monorepo with several apps using the command below, it will be installed for a default project

```sh
npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add
```

But if we add the `--project` parameter NX will understand it and run schematics only for the specified project

```sh
npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add --project=docs-app
```

I'm just not sure that the `@ng-doc/add` package should care about the project because it doesn't know anything about the environment. Some people can use the library for non-NX projects.

**Review by skoropadas — COMMENTED** [review]
Checked the PR, and I'm curious why do we need the `project` parameter? Initially, I thought this ticket was about a lack of documentation for monorepos.

So if we try to install NgDoc for monorepo with several apps using the command below, it will be installed for a default project

```sh
npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add
```

But if we add the `--project` parameter NX will understand it and run schematics only for the specified project

```sh
npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add --project=docs-app
```

I'm just not sure that the `@ng-doc/add` package should care about the project because it doesn't know anything about the environment. Some people can use the library for non-NX projects.

**Comment by eranshmil** [comment]
> Checked the PR, and I'm curious why do we need the `project` parameter? Initially, I thought this ticket was about a lack of documentation for monorepos.
> 
> So if we try to install NgDoc for monorepo with several apps using the command below, it will be installed for a default project
> 
> ```shell
> npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add
> ```
> 
> But if we add the `--project` parameter NX will understand it and run schematics only for the specified project
> 
> ```shell
> npm install @ng-doc/add && npx nx g @ng-doc/add:ng-add --project=docs-app
> ```
> 
> I'm just not sure that the `@ng-doc/add` package should care about the project because it doesn't know anything about the environment. Some people can use the library for non-NX projects.

Oops, you are right.
I had two issues while using the `add` command without specifying the project name:
1. `addAssets`	
	image

2. `addTsconfigPaths`
	image

I don't want to update the PR without asking you first, but I think that [these lines](https://github.com/ng-doc/ng-doc/blob/main/libs/add/schematics/ng-add/utils/get-project.ts#L18-L22) should be extracted to a new util function that will be reused inside `getProject` and the two steps mentioned above.
Let me know what you think and I also can update the docs about specifying the project name.

**Comment by skoropadas** [comment]
@eranshmil Once again, I apologize for the delay, I am still busy with the move.

Oh yes, I see the problem. I think the issue is that `getNgDocAssets` takes a list of arguments passed to the command. Consequently, if we don't pass the `--project` parameter, we get `undefined`.

Yes, I completely agree with you. I would also add a new function `getProjectName` and reuse it inside `getProject` and for the calls to `getNgDocAssets` and `addTsconfigPaths`. I would be happy to review your changes. 👍🏻

**Comment by eranshmil** [comment]
> @eranshmil Once again, I apologize for the delay, I am still busy with the move.
> 
> Oh yes, I see the problem. I think the issue is that `getNgDocAssets` takes a list of arguments passed to the command. Consequently, if we don't pass the `--project` parameter, we get `undefined`.
> 
> Yes, I completely agree with you. I would also add a new function `getProjectName` and reuse it inside `getProject` and for the calls to `getNgDocAssets` and `addTsconfigPaths`. I would be happy to review your changes. 👍🏻

That's fine, you don't need to be available 24/7. 
Updated the PR. :)

I will be happy to help you even more with this project. Let me know.

**Review by skoropadas — CHANGES_REQUESTED** [review]
Everything is cool, just one small comment left. Thank you for your help! ❤️ 

About help:
I still have some tasks to do for the NgDoc 18. Here are some things that you could help with:
- Implement a simple migration for guides, which involves removing `# {{ NgDocPage.title }}` that people define in their markdown files, and moving the `keyword` value into the markdown template if it exists.
- Rewrite the angular keywords loader. Previously, I used a JSON file from the angular.io assets, but after their migration to angular.dev, it's not possible anymore. So, I'll probably need to parse `https://angular.dev/api` to extract all API references for the keywords.
- Update the landing page to make it more attractive, add a features list, and some reusable components for users to easily create landing pages for their own libraries. I don't have much experience with that, so I need some help from someone who does :)
- Update the theming docs. I made some breaking changes to support SSR, so now it should be defined in a different way.
- Create docs for mermaid diagrams, including how to enable them and how to write them.

That's about it, so if you're interested in anything from this list and want to help out, just let me know.

File: libs/add/schematics/ng-add/constants/assets.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+import { getProjectName } from '../utils/get-project-name';
 
 /**
  *
  * @param options
+ * @param workspace
  */
-export function getNgDocAssets(options: Schema): Array<{
+export function getNgDocAssets(
+	options: Schema,
```
[the remark this answers is not in the document]
Comment (skoropadas) [reply libs/add/schematics/ng-add/constants/assets.ts:12]: Ah sorry, we use it for the `getProjectName` function, this is fine then!

**Review by skoropadas — APPROVED** [review]

**Pull request merged**